### PR TITLE
Bug 1836866: Fixed deployment trigger when replicas changed

### PR DESCRIPTION
**Pull request by igor-karpukhin**
The deployment will be marked as changed when the number of replicas is changed.

### CODE REVIEW

**Comment by igor-karpukhin** [comment]
@ewolinetz @jcaianirh @periklis just in case, there is a library for K8S resources comparison: https://github.com/banzaicloud/k8s-objectmatcher, how about to adopt it in the future?

File: pkg/k8shandler/kibana/reconciler.go [hunk 1]
```diff
@@ -363,6 +364,12 @@ func isDeploymentDifferent(current *apps.Deployment, desired *apps.Deployment) (
 		different = true
 	}
 
+	if current.Spec.Replicas != desired.Spec.Replicas {
```
Comment (ewolinetz) [inline pkg/k8shandler/kibana/reconciler.go:367]: i think we need to dereference this to check the values: 
*current.Spec.Replicas != *desired.Spec.Replicas

https://godoc.org/k8s.io/api/apps/v1#DeploymentSpec

Comment (jcantrill) [inline pkg/k8shandler/kibana/reconciler.go:367]: Please add a testcase for "isDeploymentDifferent"

File: pkg/k8shandler/kibana/reconciler.go [hunk 2]
```diff
@@ -363,6 +364,12 @@ func isDeploymentDifferent(current *apps.Deployment, desired *apps.Deployment) (
...
+		logrus.Debugf("Replicas changed. Previous: %d, Current: %d", current.Spec.Replicas, desired.Spec.Replicas)
```
Comment (jcantrill) [inline pkg/k8shandler/kibana/reconciler.go:368]: Please add the "kibana" instance name to the debug line given we can/could support multiples

File: pkg/k8shandler/kibana/reconciler.go (hunk 1 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/kibana/reconciler.go]: Fxied

Comment (igor-karpukhin) [reply pkg/k8shandler/kibana/reconciler.go]: fixed

File: pkg/k8shandler/kibana/reconciler.go (hunk 2 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/kibana/reconciler.go]: fixed

**Pull request merged**